app/models/group_user.rb

```diff
@@ -43,13 +43,9 @@ def remove_primary_group
     )
   end
 
-  def remove_title
-    if group.title.present?
-      DB.exec("
-        UPDATE users SET title = NULL
-        WHERE title = :title AND id = :id",
-        id: user_id, title: group.title
-      )
+  def grant_other_available_title
+    if group.title.present? && group.title == user.title
+      user.update(title: user.available_titles.first)
```

Comment:
Whenever we're not checking for the return value, we want to use `update!` so that this doesn't fail silently.